hazelcast/src/main/java/com/hazelcast/map/impl/operation/MergeBackupOperation.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.core.EntryView;
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.map.impl.MapEntries;
+import com.hazelcast.map.impl.record.Record;
+import com.hazelcast.map.impl.record.RecordInfo;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.BackupOperation;
+import com.hazelcast.spi.PartitionAwareOperation;
+import com.hazelcast.spi.merge.SplitBrainMergePolicy;
+import com.hazelcast.util.Clock;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.map.impl.EntryViews.createSimpleEntryView;
+import static com.hazelcast.map.impl.record.Records.applyRecordInfo;
+
+/**
+ * Creates backups for merged {@link MapEntries} after split-brain healing with a {@link SplitBrainMergePolicy}.
+ *
+ * @since 3.10
+ */
+public class MergeBackupOperation extends MapOperation implements PartitionAwareOperation, BackupOperation {
+
+    private MapEntries entries;
+    private List<RecordInfo> recordInfos;
+    private boolean hasWanReplication;
+
+    public MergeBackupOperation(String name, MapEntries entries, List<RecordInfo> recordInfos, boolean hasWanReplication) {
+        super(name);
+        this.entries = entries;
+        this.recordInfos = recordInfos;
+        this.hasWanReplication = hasWanReplication;
```

Comment:
Do we need to send this boolean (since it can be checked on the backup replica)?

Reply:
The merge operation gets `disableWanReplicationEvent` as parameter, so we don't just check if the WAN replication is enabled: `hasWanReplication = mapContainer.isWanReplicationEnabled() && !disableWanReplicationEvent;`. The backup operation should do the same, so I give it the `hasWanReplication` as state, so we don't lose the `disableWanReplicationEvent` part.